lib/backend/compat/compat.go

```diff
@@ -251,6 +313,44 @@ func (c *ModelAdaptor) listNodes(l model.NodeListOptions) ([]*model.KVPair, erro
 	return results, nil
 }
 
+// listBlock returns list of KVPairs for Block, includes making sure
+// backwards compatiblity. See getBlock for more details.
+func (c *ModelAdaptor) listBlock(l model.BlockListOptions) ([]*model.KVPair, error) {
+
+	// Get a list of block KVPairs.
+	bal, err := c.client.List(l)
+	if err != nil {
+		return nil, err
+	}
+
+	// Go through the list to make sure Affinity field has a proper value,
+	// and map the value to Affinity if deprecated HostAffinity field is used
+	// by calling ensureBlockAffinity, and update the KVPair slice to return.
+	for idx, bakv := range bal {
+		bal[idx] = ensureBlockAffinity(bakv)
```

Comment:
+1 to having a separate slice.
variable names are short because generally go best practice is to use shorter variable names for variables with short scope (more descriptive names as the scope grows) 
Ref: https://github.com/golang/go/wiki/CodeReviewComments#variable-names